src/transformers/models/owlvit/modeling_tf_owlvit.py

```diff
@@ -0,0 +1,1730 @@
+        output = self.gelu(output)
+        output = self.dense2(output)
+        return output
+
+
+class TFOwlViTClassPredictionHead(tf.keras.layers.Layer):
+    def __init__(self, config: OwlViTConfig, **kwargs):
+        super().__init__(**kwargs)
+
+        out_dim = config.text_config.hidden_size
+
+        self.dense0 = tf.keras.layers.Dense(units=out_dim, name="dense0")
+        self.logit_shift = tf.keras.layers.Dense(units=1, name="logit_shift")
+        self.logit_scale = tf.keras.layers.Dense(units=1, name="logit_scale")
+        self.elu = get_tf_activation("elu")
+
+    def call(
+        self,
+        image_embeds: tf.Tensor,
+        query_embeds: tf.Tensor,
+        query_mask: tf.Tensor,
+    ) -> Tuple[tf.Tensor]:
+
+        image_class_embeds = self.dense0(image_embeds)
+
+        # Normalize image and text features
+        image_class_embeds = (
+            image_class_embeds / tf.norm(tensor=image_class_embeds, ord="euclidean", axis=-1, keepdims=True) + 1e-6
+        )
+        query_embeds = query_embeds / tf.norm(tensor=query_embeds, ord="euclidean", axis=-1, keepdims=True) + 1e-6
+
+        # Get class predictions
+        pred_logits = tf.einsum("...pd,...qd->...pq", image_class_embeds, query_embeds)
+
+        # Apply a learnable shift and scale to logits
+        logit_shift = self.logit_shift(image_embeds)
+        logit_scale = self.logit_scale(image_embeds)
+        logit_scale = self.elu(logit_scale) + 1
+        pred_logits = (pred_logits + logit_shift) * logit_scale
+
+        if query_mask is not None:
+            if len(query_mask.shape) > 1:
+                query_mask = tf.expand_dims(query_mask, axis=-2)
+
+            pred_logits = tf.where(query_mask == 0, -1e6, pred_logits)
+
+        return (pred_logits, image_class_embeds)
+
+
+class TFOwlViTForObjectDetection(TFOwlViTPreTrainedModel):
+    config_class = OwlViTConfig
+
+    def __init__(self, config: OwlViTConfig, *inputs, **kwargs):
+        super().__init__(config, *inputs, **kwargs)
+
+        self.owlvit = TFOwlViTMainLayer(config, name="owlvit")
+        self.class_head = TFOwlViTClassPredictionHead(config, name="class_head")
+        self.box_head = TFOwlViTBoxPredictionHead(config, name="box_head")
+        self.layer_norm = tf.keras.layers.LayerNormalization(
+            epsilon=config.vision_config.layer_norm_eps, name="layernorm"
+        )
+
+    def normalize_grid_corner_coordinates(self, feature_map: Optional[TFModelInputType] = None):
+        # Computes normalized xy corner coordinates from feature_map.
+        if not len(feature_map.shape) == 4:
+            raise ValueError("Expected input shape is [batch_size, num_channels, height, width]")
+
+        height, width = feature_map.shape[1], feature_map.shape[2]
+
+        box_coordinates = np.stack(np.meshgrid(np.arange(1, width + 1), np.arange(1, height + 1)), axis=-1).astype(
+            np.float32
+        )
+        box_coordinates /= np.array([width, height], np.float32)
```

Comment:
It'd be easier to read (and fewer lines) if split into steps than a one-liner that has to be written across lines. 

```suggestion
        box_coordinates = np.stack(np.meshgrid(np.arange(1, width + 1), np.arange(1, height + 1)), axis=-1)
        box_coordinates = box_coordinates.astype(np.float32)
        box_coordinates /= np.array([width, height], np.float32)
```